### PR TITLE
Added standard/freeform objective toggle

### PR DESCRIPTION
**Pull request by Shaska-**
Does exactly what it says on the tin.
Allows people to choose whether they get the standard antag objectives or the new freeform objectives when they're a solo antage role.

:cl:
 * rscadd: You can now choose between standard and freeform objectives in the general settings menu

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -141,6 +141,7 @@ var/const/MAX_SAVE_SLOTS = 8
 	var/ambience_volume = 25
 	var/credits_volume = 75
 	var/window_flashing = 1
+	var/antag_objectives = 1 //If set to 1, solo antag roles will get the standard objectives. If set to 0, will give them a freeform objective instead.
```
Comment (SonixApache) [inline code/modules/client/preferences.dm]: Set this to zero, to stay in line with the recent polls.

**Comment by ArchSunder** [comment]
So why did we bother having a server poll to remove jecties if people will just spam PR's to re-add them despite the vote result

**Comment by SonixApache** [comment]
Because the poll we had showed that the opinions were extremely divided
and then someone decided to disregard that poll and thus a big portion of the playerbase
and surprise surprise, there were people that didn't like that

this is a good compromise

**Comment by ShiftyRail** [comment]
Objectives, objectives never change

File: code/modules/migrations/SS13_Prefs/021-antag_objectives.dm [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/datum/migration/sqlite/ss13_prefs/_021
+	id = 21
+	name = "Toggle Solo Antag Objectives"
+
+/datum/migration/sqlite/ss13_prefs/_021/up()
+	if(!hasColumn("client","antag_objectives"))
+		return execute("ALTER TABLE `client` ADD COLUMN antag_objectives INTEGER DEFAULT 1")
```
Comment (ShiftyRail) [inline code/modules/migrations/SS13_Prefs/021-antag_objectives.dm]: This makes people have objectives by default
While I'm not against the PR in and off itself, it really makes the poll worthless

Comment (SonixApache) [reply code/modules/migrations/SS13_Prefs/021-antag_objectives.dm]: yeah set this to 0 as well

Comment (Shaska-) [reply code/modules/migrations/SS13_Prefs/021-antag_objectives.dm]: Yeah, my bad. Worked fine locally, so I happily forgot to update the file that actually makes it work on the server as well.

**Comment by bathosbathtime** [comment]
If the objectives are already optional, what is the point in making them double optional?
While I wouldn't be opposed to re-adding classic objectives whole cloth, adding a toggle seems pointless.  The primary complaint about classic objectives were that they would encourage passive antagonizing after the individual following them completed their objectives to increase their chances of survival for the final objective (typically survive free, or hijack the shuttle).  This was on top of a few of the steal objectives being lack luster, resulting in (rare) traitors that did absolutely nothing of impact to any other player.  Adding in a toggle does nothing to address this point, and serves no benefit.
Of course, there isn't any harm in doing this this way.  I just find it to be a strange attempt at a compromise which misses the point entirely.

**Comment by SonixApache** [comment]
@bathosbathtime 
Some people enjoy getting objectives. Some people base their actions off them.
Some people enjoy seeing round-end greentext and redtext.

\>The primary complaint about classic objectives were that they would encourage passive antagonizing after the individual following them completed their objectives to increase their chances of survival for the final objective (typically survive free, or hijack the shuttle). This was on top of a few of the steal objectives being lack luster, resulting in (rare) traitors that did absolutely nothing of impact to any other player.
\>Adding in a toggle does nothing to address this point, and serves no benefit.
It's opt-in now. People that actually want objectives for reasons can get them, and if they run better without them they'll keep them off. We could even, god forbid, administratively bind people to them.

**Comment by bathosbathtime** [comment]
>Some people enjoy getting objectives. Some people base their actions off them.
Some people enjoy seeing round-end greentext and redtext.
>
Which I why I am not opposed to them being re-added.  That doesn't mean making something already optional even more optional makes any sense.
Further,  one of the good things about objectives is that they can give newer players a goal, which is undone by making it something you have to know to enable.  While it is more in line with the poll, it is poor design to have it off by default as someone that needs help with setting up their traitor roll is also someone that is less likely to be aware they have the option to enable the objectives.
There isn't anything gained by having the freeform objective available if it isn't the only objective available since we have solo-antagonist freedom anyway.

**Comment by Shaska-** [comment]
Objectives aren't currently optional, given that you can't get them at all. 
But more to the point, just having objectives can make some people feel locked into doing them, and the ability to turn them off can help them do more interesting things as an antag.
Conversely, other people can tend to feel lost without the structure provided the objectives, and end up either doing less antagging in general, as they have no goal to work towards, or go all in on murder and chaos, as there's not really any reason for them not to. 
Personally I flipflop between both positions, depending on the day.
It doesn't address the flaws inherent in the objective system, as it's not meant to do that specifically.
I think having the ability to toggle between whether or not you get objectives will make the game better for everyone, even if only marginally so for some.

**Comment by bathosbathtime** [comment]
>But more to the point, just having objectives can make some people feel locked into doing them, and the ability to turn them off can help them do more interesting things as an antag.
>
I hadn't considered that.  I find this to be a very good PR, then.
Though, I do still stick with my statement about classic being the better default.  Since it isn't in line with the poll I won't push for it to be changed, but it would be better if classic objectives were the default.  Isn't too important either way as long as the toggle is there, though.

**Comment by SonixApache** [comment]
just make another poll if this gets merged, we'll have had like 4 polls on the same subject

**Review by ShiftyRail — APPROVED** [review]
Code is good to go, will leave open a few more days as it is a major change but shouldn't have too much problems being merged.

**Comment by Kammerjunk** [comment]
let's maybe see what people want before we merge anything
there's like half a dozen different ideas on objectives currently open as PRs

does this PR remove all references to SOLO_ANTAG_OBJECTIVES? the define should probably be removed if so

**Comment by Shaska-** [comment]
Oh yeah, good point.
Can't find any other references to it, so I'll remove the define as well

**Review by MadmanMartian — APPROVED** [review]
This is the way it should have been done

**Pull request merged**